### PR TITLE
Pluribus Networks admin service module

### PR DESCRIPTION
**Pull request by rajaspachipulusu17**
##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

```

### CODE REVIEW

File: lib/ansible/modules/network/netvisor/pn_admin_service.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    required: False
+    type: str
+  state:
+    description:
+      - State the action to perform. Use 'update' to modify the admin-service.
+    required: True
+    choices: ['update']
+  pn_web:
+    description:
+      - Web (HTTP) to enable or disable
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_admin_service.py]:
```suggestion
      - Web (HTTP) to enable or disable.
```

File: lib/ansible/modules/network/netvisor/pn_admin_service.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    required: False
+    type: bool
+  pn_web_ssl:
+    description:
+      - Web SSL (HTTPS) to enable or disable
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_admin_service.py]:
```suggestion
      - Web SSL (HTTPS) to enable or disable.
```

File: lib/ansible/modules/network/netvisor/pn_admin_service.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    required: False
+    type: bool
+  pn_snmp:
+    description:
+      - Simple Network Monitoring Protocol (SNMP) to enable or disable
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_admin_service.py]:
```suggestion
      - Simple Network Monitoring Protocol (SNMP) to enable or disable.
```

File: lib/ansible/modules/network/netvisor/pn_admin_service.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    type: str
+    choices: ['mgmt', 'data']
+  pn_icmp:
+    description:
+      - Internet Message Control Protocol (ICMP) to enable or disable
+    required: False
+    type: bool
+  pn_net_api:
+    description:
+      - Netvisor API to enable or disable APIs
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_admin_service.py]:
```suggestion
      - Netvisor API to enable or disable APIs.
```

File: lib/ansible/modules/network/netvisor/pn_admin_service.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    required: False
+    type: bool
+  pn_snmp:
+    description:
+      - Simple Network Monitoring Protocol (SNMP) to enable or disable
+    required: False
+    type: bool
+  pn_web_port:
+    description:
+      - Web (HTTP) port to enable or disable
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_admin_service.py]:
```suggestion
      - Web (HTTP) port to enable or disable.
```

**Review by Akasurde — CHANGES_REQUESTED** [review]
Could you please add integration/unit test for  this change? Thanks.

File: lib/ansible/modules/network/netvisor/pn_admin_service.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: pn_admin_service
+author: "Pluribus Networks (rpachi@pluribusnetworks.com)"
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_admin_service.py]:
```suggestion
author: "Pluribus Networks (@rajaspachipulusu17)"
```

**Comment by rajaspachipulusu17** [comment]
@Akasurde , Thanks for the review. I have made all suggested changes.

**Comment by rajaspachipulusu17** [comment]
@Akasurde 

Could you please add integration/unit test for this change? Thanks.

Where should i add it. I have given example.

**Comment by Akasurde** [comment]
> Where should i add it. I have given example.

You can take a look into [this](https://github.com/ansible/ansible/tree/devel/test/units/modules/network). You can read more about [Ansible testing](https://docs.ansible.com/ansible/latest/dev_guide/testing.html) here.

**Comment by rajaspachipulusu17** [comment]
@Akasurde  it will be merged only if i send PR of unit testing?

**Comment by Akasurde** [comment]
> @Akasurde it will be merged only if i send PR of unit testing?

Yes.

**Comment by rajaspachipulusu17** [comment]
This is pushed with units in PR #49854

**Pull request closed without merging**